openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -189,6 +189,16 @@ def _submit_deadline_post_job(self, instance, job, instances):
             instances[0]["family"],
             override_version
         )
+        if instance.data.get("expected_families", []):
+            output_dir = self._get_publish_folder(
+                anatomy,
+                deepcopy(instance.data["anatomyData"]),
+                instance.data.get("asset"),
+                instance.data["subset"],
+                instance.context,
+                instance.data["family"],
+                override_version
+            )
```

Comment:
Please add a comment WHY this is needed - also, if we're overidding `output_dir` anyway this should be the `if branch` and the other logic the `else` branch so we don't compute the other for no reason.

Reply:
you need to get the correct filepath for the farm caching. otherwises it will go to the publish folder of render family.

Reply:
But the output of a pointcache is just whatever the ROP node is set to, right? What output dir is this that we need?

The way I see it all we need to do is:

- Submit a Houdini Job to Deadline
- Deadline launches Houdini and triggers OpenPype's regular "publish" (including all its validations!) but only for the instances the user has chosen on the submission.
    - It publishes and generates the output as usual, as regular publishing does.
    - Having the farm also validate the instances means it can actually confirm all loaded plug-ins still work, paths are valid, etc. - everything that we want to ensure it just doesn't go "render a rop node" and then producing different output.
    - We need almost no code changes to allow publishing on the farm.
    - Note: We don't need to keep track of any `.json` files or pre-processed data.

Or what else do we need?

Reply:
> But the output of a pointcache is just whatever the ROP node is set to, right? What output dir is this that we need?
> 
> Houdini just renders the ROP - it renders to where the ROP node is set to; and that defines our staging dir, voila. Why does the farm need to be so special about this if the farm tries to basically do what the user would otherwise do on his own machine (except for being headless)
> 
> The way I see it all we need to do is:
> 
> * Submit a Houdini Job to Deadline
> * Deadline launches Houdini and triggers OpenPype's regular "publish" (including all its validations!) but only for the instances the user has chosen on the submission.
>   
>   * It publishes and generates the output as usual, as regular publishing does.
>   * Having the farm also validate the instances means it can actually confirm all loaded plug-ins still work, paths are valid, etc. - everything that we want to ensure it just doesn't go "render a rop node" and then producing different output.
>   * We need almost no code changes to allow publishing on the farm.
>   * Note: We don't need to keep track of any `.json` files or pre-processed data.
> 
> Or what else do we need?

You need to get the latest version of the published families folder in order to get the file transmission right.
let's say `publish\pointcache\pointcacheMain\v020`

Reply:
We don't need to if the machine that "rendered the rop" itself would also publish it right.
What's the reason we explicitly want to separate on the farm the caching from the publishing [like in this screenshot](https://user-images.githubusercontent.com/20871534/265078060-9dcf1a48-d24d-409f-8129-6f2d7c07794e.png). Is that crucial?

I'm a bit wary that the implementation changes I'm seeing here will be total madness to get similar support across all families and all hosts - even though it should be as trivial as "just run the publishing on the farm without the integrating" and just separating the integrating part if we just want to have the 'publish' job separate from a cache job.